ingestion_server/ingestion_server/elasticsearch_models.py

```diff
@@ -330,12 +336,22 @@ def database_row_to_elasticsearch_doc(row, schema):
             category=row[schema["category"]],
             duration=row[schema["duration"]],
             length=length,
+            filetype=filetype,
+            extension=extension,
             authority_boost=authority_boost,
             max_boost=max(popularity or 1, authority_boost or 1),
             min_boost=min(popularity or 1, authority_boost or 1),
             **attrs,
         )
 
+    @staticmethod
+    def get_extensions(filetype, alt_files):
+        if not alt_files:
+            return filetype
+
+        extensions = [filetype] + [file["filetype"] for file in alt_files]
+        return extensions
```

Comment:
Given that the preview audio is not always available for downloading... I wonder if we should _only_ index alt_files for extensions when there are alt_files available (and filetypes in the future, etc)? :thinking: Might also be nice to always return a list instead of `str | list[str]`

```suggestion
    @staticmethod
    def get_extensions(filetype, alt_files):
        if not alt_files:
            return [filetype]

        return [file["filetype"] for file in alt_files]
```

Reply:
I tried returning always a list but when there is only a single extension it produces a parsing error.
I'm fine with including the preview's file type in the indexing. If more folks think we should not I'll exclude it.

Reply:
> I tried returning always a list but when there is only a single extension it produces a parsing error.

That's strange :thinking: I wonder what the difference could be...

> I'm fine with including the preview's file type in the indexing. If more folks think we should not I'll exclude it.

The problem is I think it might be inaccurate to include it unless we can confirm it's actually a downloadable audio track (rather than just a preview audio)... I kind of wish this list was defined in the catalog rather than munged in the API :disappointed: That way it could be per-provider and we could confirm that the list is actually the list of downloadable formats.